presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -496,6 +506,149 @@ public CounterStat getHighMemorySplitSource()
         return concat(partitionBatches);
     }
 
+    private static PartitionBatchInfo getPartitionBatchInfo(
+            ConnectorSession session,
+            SemiTransactionalHiveMetastore metastore,
+            SchemaTableName tableName,
+            List<HivePartition> partitionBatch,
+            Map<String, HiveColumnHandle> predicateColumns,
+            Optional<Map<Subfield, Domain>> domains)
+    {
+        if (isPartitionStatsBasedOptimizationEnabled(session) && domains.isPresent()) {
```

Comment:
May be can early exit if predicates do not exist or do not have non partition columns.

Reply:
Yes, we will early exit if there is no predicate domains, or there is no partition stats. The logic will be a bit different and more like the one mentioned in https://github.com/prestodb/presto/pull/15554#discussion_r563945986